module/zfs/spa_stats.c

```diff
@@ -819,6 +819,39 @@ spa_state_init(spa_t *spa)
 	kmem_strfree(name);
 }
 
+static int
+spa_guid_data(char *buf, size_t size, void *data)
+{
+	spa_t *spa = (spa_t *)data;
+	(void) snprintf(buf, size, "%llu\n", (u_longlong_t)spa_guid(spa));
+	return (0);
+}
+
+static void
+spa_guid_init(spa_t *spa)
+{
+	spa_history_kstat_t *shk = &spa->spa_stats.guid;
+	char *name;
+	kstat_t *ksp;
+
+	mutex_init(&shk->lock, NULL, MUTEX_DEFAULT, NULL);
+
+	name = kmem_asprintf("zfs/%s", spa_name(spa));
+
+	ksp = kstat_create(name, 0, "guid", "misc",
+	    KSTAT_TYPE_RAW, 0, KSTAT_FLAG_VIRTUAL);
```

Comment:
I missed this in the initial review but you need to make sure to assign `ksp` to `shk->kstat` here.  Otherwise, this won't be properly cleaned up and you'll here the [ASSERT caught by the CI](http://build.zfsonlinux.org/builders/CentOS%207%20x86_64%20%28TEST%29/builds/20256/steps/shell_4/logs/console).

Reply:
D'oh. Thanks for catching that.